dusk-labyrinth.lic

```diff
@@ -7,17 +7,15 @@ custom_require.call %w[common common-items events]
 class DuskLab
   include DRC
   include DRCI
-  def initialize
-    arg_definitions = [
-      [{ name: 'loot_bag', display: 'container noun', regex: /\w+/i, description: 'Bag for your loot!' },
-       { name: 'redeem_scrip', display: 'redeem', regex: /^redeem/i, optional: true, description: 'Redeem scrip when finished?' }]
-    ]
-    args = parse_args(arg_definitions)
+  def initialize(settings)
+    @labyrinth_data = settings.duskruin['labyrinth']
+    pet_noun = @labyrinth_data['pet_noun']
+    redeem_scrip = @labyrinth_data['redeem_scrip']
     Flags.add('five-minute-warning', 'You only have about 5 minutes left in the labyrinth!')
     Flags.add('done', 'A surly Dwarf stomps in and glowers at you in the dim gloom')
-    Flags.add('no-monkey', 'A small monkey scurries (?<rat_dir>\w+)!')
-    Flags.add('monkey', 'You notice a small monkey scurrying around the area')
-    Flags.add('caught', 'You pick them both up, claiming your new pet')
+    Flags.add('no-pet', "A small #{pet_noun} scurries #{Regexp.new('(?<pet_dir>\w+)')}!")
+    Flags.add('pet', "You notice a small #{pet_noun} scurrying around the area")
+    Flags.add('caught', "You pick them both up, claiming your new #{pet_noun}")
```

Comment:
This one used to be 'pet', are you sure it should be changed?

Reply:
```suggestion
    Flags.add('caught', 'You pick them both up, claiming your new pet')
```

Good catch, that was an oversight of mine. The logs have the following message which uses the word **pet**:

> You search around the area and find a scrawny ship rat!  The ship rat scampers toward you and nuzzles your foot, quickly deciding you're its new owner.  As you reach for the rat, it scurries into a coarse burlap sack, which is the rat home.  **You pick them both up, claiming your new pet.**
Roundtime: 4 sec.